### ISSUE TITLE
Support for create-react-app 5

### ISSUE THREAD

**Issue by shapeshifta78** [Derek-Hu/react-app-rewire-multiple-entry#31]
Will you support cra 5?

**Comment by BernhardSeybold** [comment]
+1

One thing I noticed is that the `options` object of the HtmlWebpackPlugin is renamed to `userOptions`, so you may want to expect both.

After patching this, our build ran through. :-)

**Comment by Arkellys** [comment]
Hello @BernhardSeybold, could you share your fix please?
I updated my version of CRA and I can't run my app anymore.

**Comment by BernhardSeybold** [comment]
> Hello @BernhardSeybold, could you share your fix please? I updated my version of CRA and I can't run my app anymore.

Sure, but I have to say, it's a HACK rather than a FIX. So far it seems to work for us.

```
  module.exports = {
    webpack: function(config, env) {

      // multipleEntry expects an "options" object but since cra v5 it is called "userOptions"
      // HACK -> copy userOptions reference and hope for the best
      const webpackPlugins = config.plugins.filter(p => p.constructor.name === 'HtmlWebpackPlugin');
      webpackPlugins.forEach(p => p.options = p.userOptions);

      // the original call
      multipleEntry.addMultiEntry(config);

      // now carry on with the options object
      webpackPlugins.forEach(p => { p.userOptions = p.options; delete p.options; });

      return config;
    }
  };
```

Hope it helps.

**Comment by Arkellys** [comment]
> > Hello @BernhardSeybold, could you share your fix please? I updated my version of CRA and I can't run my app anymore.
> 
> Sure, but I have to say, it's a HACK rather than a FIX. So far it seems to work for us.
> 
> ```
>   module.exports = {
>     webpack: function(config, env) {
> 
>       // multipleEntry expects an "options" object but since cra v5 it is called "userOptions"
>       // HACK -> copy userOptions reference and hope for the best
>       const webpackPlugins = config.plugins.filter(p => p.constructor.name === 'HtmlWebpackPlugin');
>       webpackPlugins.forEach(p => p.options = p.userOptions);
> 
>       // the original call
>       multipleEntry.addMultiEntry(config);
> 
>       // now carry on with the options object
>       webpackPlugins.forEach(p => { p.userOptions = p.options; delete p.options; });
> 
>       return config;
>     }
>   };
> ```
> 
> Hope it helps.

It worked, thank you!

**Comment by Derek-Hu** [comment]
Issue fixed in v2.2.2. Thanks!
